hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PaidQueryHandler.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.spi.workflows;
+
+import static com.hederahashgraph.api.proto.java.ResponseType.ANSWER_ONLY;
+import static com.hederahashgraph.api.proto.java.ResponseType.ANSWER_STATE_PROOF;
+import static com.hederahashgraph.api.proto.java.ResponseType.COST_ANSWER;
+import static java.util.Objects.requireNonNull;
+
+import com.hederahashgraph.api.proto.java.ResponseType;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * An abstract class for all queries that are not free. If payment is required depends on the {@link
+ * ResponseType}
+ */
+public abstract class PaidQueryHandler implements QueryHandler {
+
+    @Override
+    public boolean requiresNodePayment(@NonNull final ResponseType responseType) {
+        requireNonNull(responseType);
+        return responseType == ANSWER_ONLY || responseType == ANSWER_STATE_PROOF;
+    }
+
+    @Override
+    public boolean needsAnswerOnlyCost(@NonNull final ResponseType responseType) {
```

Comment:
To keep both methods similar `needsAnswerOnlyPayment` ?

Reply:
I copied the names from the original implementation. In my understanding, the names mean something slightly different, therefore having different names makes sense.

The method `requiresNodePayment()` checks, if the caller has to pay for the query. The method `needsAnswerOnlyCost()` checks, if only the costs for a query were requested.